File: test-server.js
#!/usr/bin/env node

/**
 * Quick test script to verify the MCP server works
 * Run this to check if everything is set up correctly
 */

console.log("=" .repeat(60));
console.log("Autodesk MCP Server - Test Suite");
console.log("=".repeat(60));
console.log();

let passed = 0;
let total = 0;

async function test(name, fn) {
  total++;
  console.log(`🧪 Testing: ${name}`);
  console.log("-".repeat(60));
  try {
    await fn();
    console.log(`✅ ${name} passed`);
    passed++;
  } catch (error) {
    console.log(`❌ ${name} failed: ${error.message}`);
  }
  console.log();
}

// Test 1: MCP SDK availability
await test("MCP SDK Import", async () => {
  const sdk = await import("@modelcontextprotocol/sdk/server/index.js");
  if (!sdk.Server) throw new Error("Server class not found");
});

// Test 2: Server module import
await test("Server Module", async () => {
  // Just check if the file exists and is valid JS
  const fs = await import("fs");
  const content = fs.readFileSync("./autodesk-mcp-server.js", "utf8");
  if (!content.includes("Server")) throw new Error("Server code missing");
});

// Test 3: Sample data
await test("Sample Data", async () => {
  const { SAMPLE_MODELS } = await import("./sample-data.js");
  const count = Object.keys(SAMPLE_MODELS).length;
  if (count !== 5) throw new Error(`Expected 5 models, got ${count}`);
  console.log(`   Models: ${Object.keys(SAMPLE_MODELS).join(", ")}`);
});

// Test 4: Tool handlers
await test("Tool Handlers", async () => {
  const { listTools, callTool } = await import("./tool-handlers.js");
  const tools = listTools();
  if (tools.length !== 7) throw new Error(`Expected 7 tools, got ${tools.length}`);
  console.log(`   Tools: ${tools.map(t => t.name).join(", ")}`);
});

// Test 5: Demo script
await test("Demo Script", async () => {
  const fs = await import("fs");
  const content = fs.readFileSync("./standalone-demo.js", "utf8");
  if (!content.includes("demo1_ListTools")) throw new Error("Demo functions missing");
});

console.log("=".repeat(60));
console.log("Test Summary");
console.log("=".repeat(60));
console.log(`Passed: ${passed}/${total}`);
console.log();

if (passed === total) {
  console.log("🎉 All tests passed! Server is ready to use.");
  console.log();
  console.log("Next steps:");
  console.log("1. Run the standalone demo: npm run demo");
  console.log("2. Or add to Claude Desktop config");
  console.log("3. Try asking: 'What Autodesk models do I have?'");
  process.exit(0);
} else {
  console.log("⚠️  Some tests failed. Please fix the issues above.");
  process.exit(1);
}

File: test_server.py
#!/usr/bin/env python3
"""
Quick test script to verify the MCP server works
Run this to check if the server can be initialized
"""

import json
import sys

def test_server_import():
    """Test that the server can be imported"""
    try:
        import autodesk_mcp_server
        print("✅ Server module imported successfully")
        return True
    except Exception as e:
        print(f"❌ Failed to import server: {e}")
        return False

def test_sample_data():
    """Test that sample data is available"""
    try:
        from autodesk_mcp_server import SAMPLE_MODELS
        print(f"✅ Sample data loaded: {len(SAMPLE_MODELS)} models available")
        for filename in SAMPLE_MODELS.keys():
            print(f"   - {filename}")
        return True
    except Exception as e:
        print(f"❌ Failed to load sample data: {e}")
        return False

def test_help_content():
    """Test that help content is available"""
    try:
        from autodesk_mcp_server import HELP_CONTENT
        products = list(HELP_CONTENT.keys())
        print(f"✅ Help content loaded for {len(products)} products: {', '.join(products)}")
        return True
    except Exception as e:
        print(f"❌ Failed to load help content: {e}")
        return False

def test_mcp_dependencies():
    """Test that MCP SDK is installed"""
    try:
        import mcp.server
        import mcp.types
        print("✅ MCP SDK installed and importable")
        return True
    except Exception as e:
        print(f"❌ MCP SDK not available: {e}")
        print("   Run: pip install mcp")
        return False

def main():
    print("=" * 60)
    print("Autodesk MCP Server - Test Suite")
    print("=" * 60)
    print()
    
    tests = [
        ("MCP Dependencies", test_mcp_dependencies),
        ("Server Import", test_server_import),
        ("Sample Data", test_sample_data),
        ("Help Content", test_help_content),
    ]
    
    results = []
    for test_name, test_func in tests:
        print(f"\n🧪 Testing: {test_name}")
        print("-" * 60)
        results.append(test_func())
        print()
    
    print("=" * 60)
    print("Test Summary")
    print("=" * 60)
    passed = sum(results)
    total = len(results)
    print(f"Passed: {passed}/{total}")
    
    if passed == total:
        print("\n🎉 All tests passed! Server is ready to use.")
        print("\nNext steps:")
        print("1. Add the server to your Claude Desktop config")
        print("2. Restart Claude Desktop")
        print("3. Try asking: 'What Autodesk models do I have?'")
        return 0
    else:
        print("\n⚠️  Some tests failed. Please fix the issues above.")
        return 1

if __name__ == "__main__":
    sys.exit(main())
